halo/withdraw/wrapper.go

```diff
@@ -45,12 +45,26 @@ func (w *BankWrapper) SendCoinsFromModuleToAccountNoWithdrawal(ctx context.Conte
 
 // UndelegateCoinsFromModuleToAccount intercepts all principal undelegations and
 // creates EVM withdrawal to the user account.
-// TODO(corver): This is unexpected in magellan upgrade, must be implemented in drake.
-func (w *BankWrapper) UndelegateCoinsFromModuleToAccount(ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
-	log.Error(ctx, "Unexpected call to DelegateCoinsFromAccountToModule [BUG]", nil, "sender", senderModule, "recipient", recipientAddr, "amt", amt)
+func (w *BankWrapper) UndelegateCoinsFromModuleToAccount(ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, coins sdk.Coins) error {
+	log.Debug(ctx, "Undelegating coins from module to account", "sender", senderModule, "recipient", recipientAddr, "coins", coins)
 
-	if err := w.Keeper.UndelegateCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, amt); err != nil {
-		return errors.Wrap(err, "undelegate coins from module to account")
+	if err := w.Keeper.UndelegateCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, coins); err != nil {
```

Comment:
I think we should align `UndelegateCoinsFromModuleToAccount` with `SendCoinsFromAccountToModule`, so follow the same pattern, not different patterns. 

I think we can just follow the `SendCoinsFromAccountToModule` logic for now. But also add a check for VestingAccount and error if that is the case with comment: `Current implementation doesn't support vesting accounts or locked coins". Then simply burning and creating the withdrawal in both cases is sufficient.